### PR TITLE
Blockchain: Use get_block to retrieve block

### PR DESCRIPTION
**Pull request by terencechain**
`getBlock` is better to use than `s.cfg.BeaconDB.Block` because
1. It checks DB
2. It checks the initial sync block cache
3. It returns error if the block is nil

We should be using `getblock` as much as we can in blockchain pkg

### CODE REVIEW

**Review by rauljordan — DISMISSED** [review]
Ideally the beacon DB itself is refactored to return ErrNotFound, but this is a safe approach for now

File: beacon-chain/blockchain/service.go [hunk 1]
```diff
@@ -339,14 +336,13 @@ func (s *Service) initializeHeadFromDB(ctx context.Context) error {
 				finalizedState.Slot(), flags.HeadSync.Name)
 		}
 	}
-
-	finalizedBlock, err := s.cfg.BeaconDB.Block(ctx, finalizedRoot)
-	if err != nil {
-		return errors.Wrap(err, "could not get finalized block from db")
+	if finalizedState == nil || finalizedState.IsNil() {
+		return errors.New("finalized state and block can't be nil")
```
Comment (potuz) [inline beacon-chain/blockchain/service.go:340]:
```suggestion
		return errors.New("finalized state can't be nil")
```
This is now only checking for the block.

**Review by potuz — COMMENTED** [review]
LGTM

File: beacon-chain/blockchain/service.go (hunk 1 above)
Comment (terencechain) [reply beacon-chain/blockchain/service.go:340]: thanks!

**Review by potuz — APPROVED** [review]

**Pull request merged**